src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/PackageUpdateCommandRunner.cs

```diff
@@ -278,194 +275,93 @@ bool PackageHasVulnerability(string packageId, NuGetVersion version, IReadOnlyLi
-                    var vulnerabilities = await vulnerabilityResource.GetVulnerabilityInfoAsync(cacheContext, remappedLogger, cancellationToken);
-                    return vulnerabilities;
-                }
-                return null;
-            }, cancellationToken));
-        }
-
-        List<IReadOnlyDictionary<string, IReadOnlyList<PackageVulnerabilityInfo>>> allVulnerabilities = new();
-        foreach (var task in tasks)
-        {
-            var result = await task;
-            if (result is not null)
-            {
-                if (result.KnownVulnerabilities?.Count > 0)
-                {
-                    foreach (var vulnDict in result.KnownVulnerabilities)
-                    {
-                        allVulnerabilities.Add(vulnDict);
-                    }
-                }
-            }
-        }
-
-        return allVulnerabilities;
-    }
-
-    private static async Task<LockFile> GetProjectAssetsFileAsync(
-        DependencyGraphSpec dgSpec,
-        ILoggerWithColor logger,
-        IPackageUpdateIO packageUpdateIO,
-        CancellationToken cancellationToken)
-    {
-        var previewRestoreResult = await packageUpdateIO.PreviewUpdatePackageReferenceAsync(dgSpec, new SourceCacheContext(), NullLogger.Instance, cancellationToken);
-        await packageUpdateIO.CommitAsync(previewRestoreResult, cancellationToken);
-        if (!previewRestoreResult.Success)
-        {
-            logger.LogError("Restore failed");
-            throw new NotSupportedException();
-        }
-
-        LockFile assetsFile;
-        if (previewRestoreResult.AssetsFile is not null)
-        {
-            assetsFile = previewRestoreResult.AssetsFile;
-        }
-        else
-        {
-            var packageSpec = dgSpec.GetProjectSpec(dgSpec.Restore.Single());
-            var assetsFilePath = Path.Combine(packageSpec.RestoreMetadata.OutputPath, LockFileFormat.AssetsFileName);
-            assetsFile = new LockFileFormat().Read(assetsFilePath);
-        }
-
-        return assetsFile;
-    }
-
     internal static async Task<List<PackageUpdateResult>> SelectPackagesToUpdateAsync(
         IReadOnlyList<Package> packages,
         PackageSpec project,
-        IVersionChooser versionChooser,
-        ISettings settings,
         ILoggerWithColor logger,
+        IPackageUpdateIO packageUpdateIO,
         CancellationToken cancellationToken)
     {
         if (packages is null || packages.Count == 0)
         {
-            return await SelectAllPackagesWithUpdatesAsync(project, versionChooser, settings, logger, cancellationToken);
-        }
-
-        var sourceMapping = PackageSourceMapping.GetPackageSourceMapping(settings);
-        if (sourceMapping.IsEnabled)
-        {
-            logger.LogMinimal(Strings.Unsupported_PackageSourceMapping, ConsoleColor.Red);
-            return new List<PackageUpdateResult>();
+            throw new ArgumentException("packages list must not be null or empty.", nameof(packages));
```

Comment:
I'm assuming this one is actually intentionally not localized since it's not expected to be called with an empty array.

either way I think we have some preset strings for cannot be null or empty.

Reply:
that's right. Where this method is called in `Run(..)`, it checks the packages list and when it's zero or null it calls a different method. This check and exception is kind of pointless, but it does make sure tests don't accidentally call the wrong method.

I see [a bunch of "null or empty" resources](https://github.com/search?q=repo%3ANuGet%2FNuGet.Client%20path%3A**%2F*.resx%20%22null%20or%20empty%22&type=code) in many projects in the solution, but not in this project.